### PR TITLE
Odd-Even Transposition Sort

### PR DESCRIPTION
**Pull request by CharlesRitter**
Here are single and multi-threaded implementations of Odd-Even transposition sort.
It is an O(n) sorting algorithm that performs its swaps simultaneously.

https://en.wikipedia.org/wiki/Odd%E2%80%93even_sort

### CODE REVIEW

**Comment by cclauss** [comment]
@CharlesRitter
On our GitHub Actions tests, we are getting these 45 warnings. Do you know how to fix it?
```
sorts/odd_even_transposition_parallel.py: 45 warnings
  /opt/hostedtoolcache/Python/3.12.3/x64/lib/python3.12/multiprocessing/popen_fork.py:66: DeprecationWarning: This process (pid=1839) is multi-threaded, use of fork() may lead to deadlocks in the child.
    self.pid = os.fork()
```

File: sorts/Odd-Even_transposition_parallel.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+This implementation represents each variable in the list with a process and
+each process communicates with its neighboring processes in the list to perform
+comparisons.
+They are synchronized with locks and message passing but other forms of
+synchronization could be used.
+"""
+from multiprocessing import Process, Pipe, Lock
+
+#lock used to ensure that two processes do not access a pipe at the same time
+processLock = Lock()
```
Comment (XuehaiPan) [inline sorts/Odd-Even_transposition_parallel.py:16]: @cclauss We can use the `spawn` method to start the subprocess. The `spawn` method is considered safer than the `fork` method. [We will get a warning for the `fork` method in Python 3.12+.](https://docs.python.org/3/library/multiprocessing.html#contexts-and-start-methods)

```suggestion
from multiprocessing import Process, Pipe, Lock, set_start_method

set_start_method("spawn", force=True)  # set the start method before using the context in `Lock()` below

#lock used to ensure that two processes do not access a pipe at the same time
processLock = Lock()
```

Comment (cclauss) [reply sorts/Odd-Even_transposition_parallel.py:16]: Nice!   Can you please make this a pull request?

Comment (XuehaiPan) [reply sorts/Odd-Even_transposition_parallel.py:16]: I opened a PR to resolve this.

- #11391

**Pull request merged**